### DISCUSSION TITLE
linking issue for iOS latest realm-js

### DISCUSSION THREAD

**Discussion by stevenmathers** [realm/realm-js#5506] [Q&A]
Error shown below. The thing is, it builds and runs fine locally using xcode, but when I use App Center to build it, it breaks.

xcode version 13.4.1 node version 16, react native 71.3, realm-js 11.5.1

where should I look?

thanks

======

 Undefined symbols for architecture arm64: 
"_objc_msgSend$allBundles", referenced from: 
realm::copy_bundled_realm_files() in librealm-js-ios.a(platform.o) 

and a bunch more like this...

 ld: symbol(s) not found for architecture arm64 
clang: error: linker command failed with exit code 1 (use -v to see invocation) 
ARCHIVE FAILED

**Comment by haswalt** [comment]
I get a similar error with xCode 14:

```
Undefined symbol: realm::set_default_realm_file_directory(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)

```

For your issue I did find a bug report that says upgrading to xcode 14 fixes it.

**Comment by sivaprasadkv** [comment]
I am getting a similar error with Xcode 13.4.1, Realm v12.6.2, React Native v0.73.5, and iOS SDK 14.0 as deployment target. While building the app with Xcode, gets this error and Xcode build logs as follows:

Undefined symbols for architecture arm64:
  "_objc_msgSend$allBundles", referenced from:
      realm::JsPlatformHelpers::copy_bundled_realm_files() in librealm-js-ios.a(platform.o)
  "_objc_msgSend$countByEnumeratingWithState:objects:count:", referenced from:
      realm::JsPlatformHelpers::copy_bundled_realm_files() in librealm-js-ios.a(platform.o)
      realm::JsPlatformHelpers::remove_realm_files_from_directory(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in librealm-js-ios.a(platform.o)
  "_objc_msgSend$stringByAppendingPathComponent:", referenced from:
      realm::JsPlatformHelpers::copy_bundled_realm_files() in librealm-js-ios.a(platform.o)
      realm::JsPlatformHelpers::remove_realm_files_from_directory(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in librealm-js-ios.a(platform.o)
  "_objc_msgSend$enumeratorAtPath:", referenced from:
      realm::JsPlatformHelpers::copy_bundled_realm_files() in librealm-js-ios.a(platform.o)
      realm::JsPlatformHelpers::remove_realm_files_from_directory(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in librealm-js-ios.a(platform.o)
  "_objc_msgSend$copyItemAtPath:toPath:error:", referenced from:
      realm::JsPlatformHelpers::copy_bundled_realm_files() in librealm-js-ios.a(platform.o)
  "_objc_msgSend$userInfo", referenced from:
      error_description(NSError*) in librealm-js-ios.a(platform.o)
  "_objc_msgSend$pathExtension", referenced from:
      realm::JsPlatformHelpers::remove_realm_files_from_directory(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in librealm-js-ios.a(platform.o)
  "_objc_msgSend$isEqualToString:", referenced from:
      realm::JsPlatformHelpers::remove_realm_files_from_directory(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in librealm-js-ios.a(platform.o)
  "_objc_msgSend$stringWithUTF8String:", referenced from:
      realm::JsPlatformHelpers::ensure_directory_exists_for_file(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in librealm-js-ios.a(platform.o)
      realm::JsPlatformHelpers::copy_bundled_realm_files() in librealm-js-ios.a(platform.o)
      realm::JsPlatformHelpers::remove_realm_files_from_directory(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in librealm-js-ios.a(platform.o)
      realm::JsPlatformHelpers::remove_file(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in librealm-js-ios.a(platform.o)
  "_objc_msgSend$code", referenced from:
      realm::JsPlatformHelpers::remove_file(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in librealm-js-ios.a(platform.o)
  "_objc_msgSend$createDirectoryAtPath:withIntermediateDirectories:attributes:error:", referenced from:
      realm::JsPlatformHelpers::ensure_directory_exists_for_file(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in librealm-js-ios.a(platform.o)
  "_objc_msgSend$objectForKeyedSubscript:", referenced from:
      error_description(NSError*) in librealm-js-ios.a(platform.o)
  "_objc_msgSend$containsString:", referenced from:
      realm::JsPlatformHelpers::copy_bundled_realm_files() in librealm-js-ios.a(platform.o)
  "_objc_msgSend$defaultManager", referenced from:
      realm::JsPlatformHelpers::ensure_directory_exists_for_file(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in librealm-js-ios.a(platform.o)
      realm::JsPlatformHelpers::copy_bundled_realm_files() in librealm-js-ios.a(platform.o)
      realm::JsPlatformHelpers::remove_realm_files_from_directory(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in librealm-js-ios.a(platform.o)
      realm::JsPlatformHelpers::remove_file(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in librealm-js-ios.a(platform.o)
  "_objc_msgSend$stringByDeletingLastPathComponent", referenced from:
      realm::JsPlatformHelpers::ensure_directory_exists_for_file(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in librealm-js-ios.a(platform.o)
  "_objc_msgSend$fileExistsAtPath:", referenced from:
      realm::JsPlatformHelpers::ensure_directory_exists_for_file(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in librealm-js-ios.a(platform.o)
      realm::JsPlatformHelpers::copy_bundled_realm_files() in librealm-js-ios.a(platform.o)
  "_objc_msgSend$removeItemAtPath:error:", referenced from:
      realm::JsPlatformHelpers::remove_realm_files_from_directory(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in librealm-js-ios.a(platform.o)
      realm::JsPlatformHelpers::remove_file(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in librealm-js-ios.a(platform.o)
  "_objc_msgSend$localizedDescription", referenced from:
      error_description(NSError*) in librealm-js-ios.a(platform.o)
  "_objc_msgSend$objectAtIndexedSubscript:", referenced from:
      realm::JsPlatformHelpers::default_realm_file_directory() in librealm-js-ios.a(platform.o)
  "_objc_msgSend$resourcePath", referenced from:
      realm::JsPlatformHelpers::copy_bundled_realm_files() in librealm-js-ios.a(platform.o)
  "_objc_msgSend$UTF8String", referenced from:
      realm::JsPlatformHelpers::default_realm_file_directory() in librealm-js-ios.a(platform.o)
      realm::JsPlatformHelpers::ensure_directory_exists_for_file(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in librealm-js-ios.a(platform.o)
      realm::JsPlatformHelpers::copy_bundled_realm_files() in librealm-js-ios.a(platform.o)
      realm::JsPlatformHelpers::remove_realm_files_from_directory(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in librealm-js-ios.a(platform.o)
      realm::JsPlatformHelpers::remove_file(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in librealm-js-ios.a(platform.o)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

<img width="704" alt="Realm_React_Native_iOS_link_error" src="https://github.com/realm/realm-js/assets/3582975/0c68b84b-6007-4800-8f5d-8d2fda274e0d">

What is the solutions for this linking issue? Please suggest.
